### PR TITLE
ResetParameters when Scenario parameters component is mount

### PR DESCRIPTION
**Pull request by jreynard-code**
fix: resetParameters when Scenario parameters component is mount

### CODE REVIEW

File: src/components/ScenarioParameters/ScenarioParameters.js [hunk 1]
```diff
@@ -133,6 +92,33 @@ const ScenarioParameters = ({
...
+  const [restockQuantity, setRestockQuantity] = useState();
+  const [waitersNumber, setWaitersNumber] = useState();
+
+  // State for basic input types examples defaultScenarioParameters
+  const [currency, setCurrency] = useState();
+  const [currencyName, setCurrencyName] = useState();
+  const [currencyValue, setCurrencyValue] = useState();
+  const [currencyUsed, setCurrencyUsed] = useState();
+  const [startDate, setStartDate] = useState();
+
```
Comment (csm-thu) [inline src/components/ScenarioParameters/ScenarioParameters.js:121]: Not initializing the states on creation leads to many errors in the browser console such as "Failed prop type: The prop `stock` is marked as required in `BasicTypes`, but its value is `undefined`."

**Review by csm-thu — CHANGES_REQUESTED** [review]
This commit seems to break parameters initialization for existing scenarios (for instance, you can check scenario "THU - Scenario with basic types parameters 2", the text field should contain "EUROTEST" but the default value is displayed)

File: src/components/ScenarioParameters/ScenarioParameters.js [hunk 2]
```diff
@@ -72,58 +101,33 @@ const ScenarioParameters = ({
...
   useEffect(() => {
-    defaultScenarioParameters.current = currentScenario.data.parametersValues;
+    const scenarioParameters = currentScenario.data.parametersValues;
+    defaultScenarioParameters.current = scenarioParameters;
+    resetParameters(false, scenarioParameters);
     /// //////////////////////////////////////////////////////////////////////// INITIAL STOCK
     const initialStockParameter = currentScenario.data?.parametersValues?.find(el => el.parameterId === INITIAL_STOCK_PARAM_ID);
     setInitialStockDatasetId(initialStockParameter?.value === undefined ? '' : initialStockParameter.value);
     /// //////////////////////////////////////////////////////////////////////// INITIAL STOCK
+  }, [currentScenario]);
```
Comment (csm-thu) [inline src/components/ScenarioParameters/ScenarioParameters.js:111]: There are eslint warnings on this line

**Review by csm-thu — APPROVED** [review]
LGTM

File: src/components/ScenarioParameters/ScenarioParameters.js (hunk 1 above)
Comment (jreynard-code) [reply src/components/ScenarioParameters/ScenarioParameters.js]: :+1:

**Pull request merged**